src/components/ec-smart-table/ec-smart-table.vue

```diff
@@ -260,4 +280,38 @@ function getEcTableSlots() {
   delete tableSlots.pages;
   return tableSlots;
 }
+
+// infinite scroll
+
+const tableEndDetector = ref(null);
+
+const canLoadMore = computed(() => props.data && props.data.length < props.totalRecords);
+
+onMounted(() => {
+  const { stop: stopIntersectionObserver } = useIntersectionObserver(
+    tableEndDetector,
+    /* c8 ignore start */
+    ([{ isIntersecting }]) => {
+      if (isIntersecting && !props.isFetching) {
+        page.value += 1;
+      }
+    },
+    /* c8 ignore stop */
+  );
+
+  watch(() => canLoadMore.value, () => {
+    if (!canLoadMore.value) {
```

Comment:
what if it is changed from false to true? Should we resume IntersectionObserver again?

Reply:
Not in the current implementation because when `canLoadMore.value` is false (basically when we already fetch all data) we just completely stop the IntersectionObserver as it is no longer needed

If in future tasks we don't want that behavior we can pause and resume the observer by using [`pause` and `resume`](https://github.com/vueuse/vueuse/blob/main/packages/core/useIntersectionObserver/index.ts#L112) methods provided by the `useIntersectionObserver` composable instead